atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLog.java

```diff
@@ -169,9 +169,7 @@ public void deleteAfter(final long index) {
   }
 
   public void flush() {
-    if (flushExplicitly) {
```

Comment:
> All callers of this method check if it should explicit flush. So another check here is not necessary. Besides, for forceFlush we have to ensure that it overrides shouldFlushExplicitly flag.

:thinking: Not sure why we have this check tbh, but sounds like you had also do it the otherway around? Like just having one check instead of everywhere in the callers? 

:question: BTW can we remove the property `flushExplicity` now?

Reply:
> Not sure why we have this check tbh

Didn't understand your comment. This check is not needed here because the callers are always checking this. So this was duplicate and unnecessary. In addition, now we want to ensure the it flushes even flushExplicitly is false because we are force flushing it.

Reply:
>  BTW can we remove the property flushExplicity now?

Not yet. Raft roles uses this flag to decide if it should flush log before commit/ack.